sdk/core/core/inc/az_curl_adapter.h

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_ADAPTER
+#define AZ_CURL_ADAPTER
+
+#include <stdio.h>
+#include <curl/curl.h>
+#include <az_span.h>
+
+CURL *curl;
+
+az_result az_http_client_init_impl() {
+    curl = curl_easy_init();
+    curl_easy_setopt(curl, CURLOPT_FAILONERROR, 1l);
+    return AZ_OK;
+}
+
+az_result az_http_client_set_URL_impl(az_const_span span) {
+    curl_easy_setopt(curl, CURLOPT_URL, span.begin);
```

Comment:
You can't assume that span.begin points to a zero-terminated string. Use _to_c_str(), it is in master now.